src/helper.ts

```diff
@@ -84,59 +135,150 @@ export function saveFilesIntoEmscriptenFS(
   }
 }
 
+export interface IUntarCondaPackageOptions {
+  /**
+   * The URL to the package
+   */
+  url: string;
+
+  /**
+   * The current untarjs instance
+   */
+  untarjs: IUnpackJSAPI;
+
+  /**
+   * Whether the functino will be verbose or not
+   */
+  verbose?: boolean;
+
+  /**
+   * Whether or not to generate conda-meta files
+   */
+  generateCondaMeta?: boolean;
+
+  /**
+   * The prefix for relocation
+   */
+  relocatePrefix?: string;
+
+  /**
+   * The environment Python version, if it is there
+   */
+  pythonVersion?: number[];
+}
+
 /**
  * Untar conda or empacked package, given a URL to it. This will also do prefix relocation.
- * @param url The URL to the package
- * @param untarjs The current untarjs instance
- * @param verbose Whether it's verbose or not
- * @param generateCondaMeta Whether or not to generate conda meta files
+ * @param options The functino options
  * @returns the files to install
  */
 export async function untarCondaPackage(
-  url: string,
-  untarjs: IUnpackJSAPI,
-  verbose = false,
-  generateCondaMeta = false
+  options: IUntarCondaPackageOptions
 ): Promise<FilesData> {
+  const {
+    url,
+    untarjs,
+    verbose,
+    generateCondaMeta,
+    relocatePrefix,
+    pythonVersion
+  } = options;
+
   const extractedFiles = await untarjs.extract(url);
 
-  if (Object.keys(extractedFiles).length !== 0) {
-    if (url.toLowerCase().endsWith('.conda')) {
-      let condaPackage: Uint8Array = new Uint8Array();
-      let packageInfo: Uint8Array = new Uint8Array();
+  const { info, pkg } = await splitPackageInfo(url, extractedFiles, untarjs);
 
-      Object.keys(extractedFiles).map(file => {
-        if (file.startsWith('pkg-')) {
-          condaPackage = extractedFiles[file];
-        } else if (file.startsWith('info-')) {
-          packageInfo = extractedFiles[file];
-        }
-      });
+  // Prefix relocation
+  if (info['info/paths.json']) {
+    const paths = JSON.parse(
+      new TextDecoder('utf-8').decode(info['info/paths.json'])
+    );
+    for (const filedesc of paths['paths']) {
+      if (!filedesc['prefix_placeholder']) {
+        continue;
+      }
 
-      if (
-        (condaPackage && condaPackage.byteLength === 0) ||
-        (packageInfo && packageInfo.byteLength === 0)
-      ) {
-        throw new Error(`Invalid .conda package ${url}`);
+      const prefixPlaceholder = filedesc['prefix_placeholder'].endsWith('/')
```

Comment:
Merely looking if the `prefix_placeholder` entry is there for files listed in "info/paths.json" to see if they need relocation.